src/planscape/planning/views.py

```diff
@@ -182,3 +181,196 @@ def list_plans(request: HttpRequest) -> HttpResponse:
+# This also can return the result as well, optionally.
+def get_scenario_by_id(request: HttpRequest) -> HttpResponse:
+    try:
+        user = _get_user(request)
+        if user is None:
+            raise ValueError("User must be logged in.")
+
+        show_results = request.GET.get('show_results', False)
+
+        scenario = Scenario.objects.select_related('planning_area__user').get(id=request.GET['id'])
+        if (scenario.planning_area.user.pk != user.pk):
+            raise ValueError("Scenario not found.")
+
+        scenario_result = None
+        if show_results:
+            scenario_result = ScenarioResult.objects.get(scenario__id=scenario.pk)
+
+        return JsonResponse(
+            _serialize_scenario(scenario, scenario_result),
+            safe=False)
+    except Exception as e:
+        return HttpResponseBadRequest("Ill-formed request: " + str(e))
+
+# Requires a logged in user, a plan_id, and a configuration for the new scenario.
+# This creates both the scenario and a corresponding ScenarioResult entry for that scenario.
+# Note that tehe ScenarioResult is by design the default one; you cannot insert a new ScenarioResult with
+# custom values.
+def create_scenario(request: HttpRequest) -> HttpResponse:
+    try:
+        # Check that the user is logged in.
+        user = _get_user(request)
+        if user is None:
+            raise ValueError("User must be logged in.")
+
+        body = json.loads(request.body)
+    
+        plan = get_object_or_404(user.planning_areas, id=body['plan_id'])
+        
+        configuration = body.get('configuration')
+        name = body.get('name')
+        if name is None:
+            raise ValueError('Must specify a scenario name.')
+
+        if configuration is None:
+            raiseValueErrors('Must provide a configuration')
+
+        # TODO: Parse configuration field into further components.
+
+        scenario = Scenario.objects.create(
+            planning_area=plan, name=name, configuration=configuration)
+        scenario.save()
+
+        # Create a default scenario result.
+        # Note that if this fails, we will have still written the Scenario without
+        # a corresponding ScenarioResult.
+        scenario_result = ScenarioResult.objects.create(
+            scenario=scenario)
+        scenario_result.save()
+
+        return HttpResponse(str(scenario.pk))
+    except Exception as e:
+        return HttpResponseBadRequest("Ill-formed request: " + str(e))
+
+#TODO
+#def list_results_for_scenario(request: HttpRequest) -> HttpResponse:
+
+# TODO: add more things to update other than state
+# Can update a state only such that pending -> running | failure, and running -> success | failure
+# Throws an error if no scenario owned by the user can be found with the desired ID.
+def update_scenario_result(request: HttpRequest) -> HttpResponse:
+    try:
+        # Check that the user is logged in.
+        user = _get_user(request)
```

Comment:
Would there be a user associated with this request? (Aka is this a request that a user would make? Or is this called completely from the backend?)

Reply:
Wow.  This is a really good catch, as the EP won't have a user login context.  Updated, but we'll have a TODO in order to lock down access to just the EP and tests.